web/src/alert_words_ui.ts

```diff
@@ -50,12 +51,19 @@ const open_alert_word_status_banner = (alert_word: string, is_error: boolean): v
         custom_classes: "alert-word-status-banner",
     };
     if (is_error) {
-        alert_word_status_banner.label = $t({defaultMessage: "Error removing alert word!"});
+        alert_word_status_banner.label = new Handlebars.SafeString(
+            $t_html(
+                {defaultMessage: "Error removing alert word <z-alert-word></z-alert-word>!"},
+                {"z-alert-word": () => `<b>${alert_word}</b>`},
+            ),
+        );
         alert_word_status_banner.intent = "danger";
     } else {
-        alert_word_status_banner.label = $t(
-            {defaultMessage: `Alert word "{alert_word}" removed successfully!`},
-            {alert_word},
+        alert_word_status_banner.label = new Handlebars.SafeString(
```

Comment:
You can't just tag a user-provided input string as safe in this way -- it's an XSS vulnerability to do so. It's critical to be paranoid about this -- whenever you're constructing strings and then putting them into the DOM, if you're using anything outside our standard patterns that manage this issue for you, you should (1) make sure you understand it and (2) Test to see if HTML in the user-provided string is actually being escaped.

Removing an alert word of the form `<script>window.open("https://google.com")</script>` will open that in a browser.

Reply:
Apologies, I overlooked that, but I'll keep a mental note of this going forward and make sure it is in my checklist to verify any XSS vulnerabilities while working on user-provided strings.

Reply:
This works, but we can use these APIs better ans skip the explicit escaping.